test/integration/targets/nxos_vlan/tests/common/interface.yaml

```diff
@@ -58,6 +61,32 @@
     that:
       - 'result.changed == false'
 
+- name: Check interfaces intent
+  nxos_vlan:
+    vlan_id: 100
+    associated_interfaces:
+      - "{{ testint1 }}"
+      - "{{ testint2 }}"
+    provider: "{{ connection }}"
+  register: result
+
+- assert:
+    that:
+      - "result.failed == false"
+
+- name: Check interfaces intent fail
+  nxos_vlan:
+    vlan_id: 100
+    associated_interfaces:
+      - test
+    provider: "{{ connection }}"
+  register: result
+  ignore_errors: yes
+
+- assert:
+    that:
+      - "result.failed == True"
```

Comment:
The nxos test is failing here. the above intent task is supposed to fail, but it is passing.
```
TASK [nxos_vlan : assert] 
fatal: [nxos9k-02]: FAILED! => {
    "assertion": "result.failed == True", 
    "changed": false, 
    "evaluated_to": false, 
    "failed": true
}
```

I did some debugging. Looks like the `associated_interfaces` key is getting deleted somewhere.
```
map_params_to_obj: 
      obj=[{'admin_state': 'up', 'associated_interfaces': ['test'], 'interfaces': None, 'mapped_vni': 'None', 'mode': 'ce', 'name': None, 'state': 'present', 'vlan_id': '100', 'vlan_state': 'active'}]
check_declarative_intent_params: 
      w={'admin_state': 'up', 'mapped_vni': 'None', 'mode': None, 'vlan_id': '100', 'vlan_state': 'active'}
```

Reply:
Fixed